processor/k8sprocessor/processor.go

```diff
@@ -75,12 +76,18 @@ func (kp *kubernetesprocessor) GetCapabilities() processor.Capabilities {
 }
 
 func (kp *kubernetesprocessor) Start(host component.Host) error {
-	go kp.kc.Start()
+	if kp.kc != nil {
+		go kp.kc.Start()
+	} else if !kp.passthroughMode {
+		return errors.New("KubeClient not initialized and not in passthrough mode")
+	}
```

Comment:
LGTM. May be only check we should have is to test if passthroughMode is enabled or not and let it crash otherwise if the client is not initialized? Your call. Can merge as is as far as I'm concerned.

Reply:
Sounds good, just updated it